From cd39162bb91980974ba13bf7b9a9507f96664532 Mon Sep 17 00:00:00 2001
From: Luciana Fujii Pontello <luciana.fujii@collabora.co.uk>
Date: Sun, 25 Sep 2011 19:04:24 -0300
Subject: [PATCH 094/119] pvrvideosink: xevent handling of internal xwindow

Handle expose and resize for when the window is created internally.
---
 sys/pvr2d/gstpvrvideosink.c |    6 ++----
 sys/pvr2d/gstpvrvideosink.h |    1 -
 2 files changed, 2 insertions(+), 5 deletions(-)

diff --git a/sys/pvr2d/gstpvrvideosink.c b/sys/pvr2d/gstpvrvideosink.c
index 47bbd52..c0f5e9c 100644
--- a/sys/pvr2d/gstpvrvideosink.c
+++ b/sys/pvr2d/gstpvrvideosink.c
@@ -407,12 +407,10 @@ gst_pvrvideosink_create_window (GstPVRVideoSink * pvrvideosink, gint width,
   g_mutex_lock (pvrvideosink->x_lock);
 
   root = DefaultRootWindow (dcontext->x_display);
-  xwindow->window_attributes.event_mask =
-      ExposureMask | PointerMotionMask | KeyPressMask;
-
   xwindow->window = XCreateSimpleWindow (dcontext->x_display, root, 0, 0,
       width, height, 2, 2, pvrvideosink->dcontext->black);
-
+  XSelectInput (dcontext->x_display, xwindow->window,
+      ExposureMask | StructureNotifyMask);
   XMapWindow (dcontext->x_display, xwindow->window);
 
   /* We have to do that to prevent X from redrawing the background on
diff --git a/sys/pvr2d/gstpvrvideosink.h b/sys/pvr2d/gstpvrvideosink.h
index 1826f9f..0100b84 100644
--- a/sys/pvr2d/gstpvrvideosink.h
+++ b/sys/pvr2d/gstpvrvideosink.h
@@ -85,7 +85,6 @@ struct _GstDrawContext
 struct _GstXWindow
 {
   Window window;
-  XSetWindowAttributes window_attributes;
   gint width, height;
   gboolean internal;
   GC gc;
-- 
1.7.5.4

